02-peer-protocol.md

```diff
@@ -952,22 +952,23 @@ Otherwise, on reconnection, a node MUST transmit `channel_reestablish`
 for each channel, and MUST wait for to receive the other node's
 `channel_reestablish` message before sending any other messages for
 that channel.  The sending node MUST set `commitments_received` to the
-number of `commitment_signed` messages received, and MUST set
-`revocations_received` to the number of `revoke_and_ack` messages
-received.
-
-If `commitments_received` is one less than the number of
-`commitment_signed` messages the receiving node has sent, it MUST send
-a set of `update_` messages against the prior commitment transaction
-followed by `commitment_signed`, otherwise if `commitments_received`
-is not equal to the number of `commitment_signed` messages the
-receiving node has sent, it SHOULD fail the channel.
-
-If `revocations_received` is one less than the number of
-`revoke_and_ack` messages the receiving node has sent, it MUST re-send
+commitment number of the last `commitment_signed` it received, and
+MUST set `revocations_received` to the commitment number of the last
+`revoke_and_ack` message received.
+
+If `commitments_received` is one less than the commitment number of
+the last `commitment_signed` message the receiving node has sent, it
+MUST send a set of `update_` messages against the prior commitment
+transaction followed by `commitment_signed`, otherwise if
+`commitments_received` is not equal to the commitment number of the
+last `commitment_signed` message the receiving node has sent, it
+SHOULD fail the channel.
+
```

Comment:
You're right.  I'll fix this using your wording.

Now, in practice, you have to remember the state when you send commitment_signed, so you probably just retransmit (that's what I do).

Re: expired HTLCs.  The receiver will fail with expiry_too_soon, yes.  But note that there's only a narrow window where sender would send them: they had to set their deadline when sending the first commitment_signed message, and that would have expired and forced them to go on-chain.  Otherwise, they might have received the commitment_signed, and published the commitment transaction containing the expired HTLC then fulfilled.